tests/utils/stl/test/file_parsing.py

```diff
@@ -0,0 +1,153 @@
+from collections import namedtuple
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import Dict, FrozenSet, List, Optional, Set, Tuple, Union
+import itertools
+import os
+import re
+
+import lit.Test
+
+_envlst_cache = dict()
+_preprocessed_file_cache = dict()
+_expected_result_entry_cache = dict()
+
+
+@dataclass
+class _TmpEnvEntry:
+    tags: Set[str] = field(default_factory=set)
+    env: Dict[str, str] = field(default_factory=dict)
+
+
+@dataclass(frozen=True)
+class EnvEntry:
+    def __init__(self, tmp_env: _TmpEnvEntry):
+        object.__setattr__(self, "_tags", frozenset(tmp_env.tags))
+        object.__setattr__(self, "_env_keys", tuple(tmp_env.env.keys()))
+        object.__setattr__(self, "_env_vals", tuple(tmp_env.env.values()))
+
+    def getEnvVal(self, key: str, default: Optional[str] = None)\
+            -> Optional[str]:
+        if key not in self._env_keys:
+            return default
+
+        # TODO: All of this is to avoid having to install frozendict.
+        # Reconsider this at a future date.
+        return self._env_vals[self._env_keys.index(key)]
+
+    def hasTag(self, tag: str) -> bool:
+        return tag in self._env_entry.tags
+
+    _tags: FrozenSet[str]
+    _env_keys: Tuple[str]
+    _env_vals: Tuple[str]
+
+
+@dataclass

```

Comment:
ahh this is my code so clearly it's prefect (other people please review it for confusing things)

Reply:
There are changes here that I think would be worth you looking over but yeah, ~80% of this is you.